### PR TITLE
Implements Tax System 

### PR DESCRIPTION
**Pull request by codestripper**
This PR meets 2/3 requirements of #1 

Edit:
This PR now fulfills all 3 requirements and is now based on the new tax system defined in the "Horizons Towns Design" -> "Town Taxes" section.

### CODE REVIEW

**Comment by LJNIC** [comment]
Could you move the `TaxTimer` stuff to just be in the `TownService` maybe?

**Comment by LJNIC** [comment]
Also, we'll probably wait on this until it's decided how taxes will be calculated.

**Comment by codestripper** [comment]
> Could you move the `TaxTimer` stuff to just be in the `TownService` maybe?

This has been completed.

**Comment by codestripper** [comment]
Hit the wrong button ^

File: src/main/java/com/atherys/towns/service/TownService.java [hunk 1]
```diff
@@ -373,4 +418,75 @@ public void removeResidentFromTown(User user, Resident resident, Town town) {
...
+            Task.Builder taxTimer = Task.builder();
+            taxTimer.interval(config.TAXES.TAX_COLLECTION_TIMER_MINUTES, TimeUnit.MINUTES)
+                    .execute(TaxTimerTask())
+                    .submit(AtherysTowns.getInstance());
+        }
+    }
+
+    private double getTaxAmount(Town town) {
+        long townSize = town.getResidents().stream()
+                .filter(resident -> Duration.between(resident.getLastLogin(), LocalDateTime.now()).compareTo(Duration.ofDays(14)) < 0)
```
[the remark this answers is not in the document]
Comment (LJNIC) [reply src/main/java/com/atherys/towns/service/TownService.java:433]: +1 on making it config based.

[the remark this answers is not in the document]
Comment (codestripper) [reply src/main/java/com/atherys/towns/service/TownService.java]: This has been resolved.

File: src/main/java/com/atherys/towns/model/entity/Town.java [hunk 2]
```diff
@@ -63,6 +67,16 @@
 
     private UUID bank;
 
+    private LocalDateTime lastTaxDate;
+
+    private int taxFailedCount;
+
+    private boolean plotClaimingDisabled;
+
+    private boolean pvpToggleDisabled;
```
[the remark this answers is not in the document]
Comment (codestripper) [reply src/main/java/com/atherys/towns/model/entity/Town.java]: This has been fixed.

File: src/main/java/com/atherys/towns/api/permission/town/TownPermissions.java [hunk 3]
```diff
@@ -63,4 +63,7 @@
 
     // permission to destroy the town
     public static final TownPermission RUIN_TOWN = new TownPermission("atherystowns.town.ruin", "Ruin Town");
+
+    // permission to pay off taxes
+    public static final TownPermission DEBT_PAY = new TownPermission("atherys.town.debt.pay", "Pay back debt");
```
Comment (AaronFoley) [inline src/main/java/com/atherys/towns/api/permission/town/TownPermissions.java:68]: Should probably follow the standard, maybe:
atherystowns.town.pay_debt

Comment (codestripper) [reply src/main/java/com/atherys/towns/api/permission/town/TownPermissions.java:68]: oh, i'm not sure why I split up atherys and town like that lol, good catch.

File: src/main/java/com/atherys/towns/api/permission/town/TownPermissions.java [hunk 4]
```diff
@@ -63,4 +63,7 @@
 
     // permission to destroy the town
     public static final TownPermission RUIN_TOWN = new TownPermission("atherystowns.town.ruin", "Ruin Town");
+
+    // permission to pay off taxes
+    public static final TownPermission DEBT_PAY = new TownPermission("atherystowns.town.pay_debt", "Pay back debt");
```
Comment (AaronFoley) [inline src/main/java/com/atherys/towns/api/permission/town/TownPermissions.java:68]: Think this also needs to go into PermissionRegistryModule

**Pull request merged**